[quant] conv_transpose graph patterns

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43199 [quant] conv_transpose graph patterns**
* #44844 [quant] Fix ConvTranspose mapping

Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

Commit:
[quant] conv_transpose graph patterns

[ghstack-poisoned]

jerryzh168 reviewed (commented):
could you add a test for this as well

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

z-a-f commented on test/quantization/test_quantize_jit.py:
```diff
@@ -3026,6 +3026,36 @@ def test_conv(self):
+                    inplace=False)
+                self.assertEqual(model_quantized(self.img_data_2d[0][0]),
+                                 result_eager,
+                                 rtol=1e-2,
+                                 atol=1e-2)
+
```
@jerryzh168 For some reason there is no exact match -- 1e-2 works though :/

jerryzh168 replied:
can you show is it the traced model or scripted model that is failing?

z-a-f replied:
It's failing for both trace model and the script model

z-a-f replied:
Oh I think the graph is not even replaced with the quantized versions -- I might have missed something

jerryzh168 replied:
yeah, please print the graphs

z-a-f replied:
> yeah, please print the graphs

```console
Eager model AnnotatedConvModel(
  (conv): QuantizedConvTranpose2d(3, 5, kernel_size=(3, 3), stride=(1, 1), scale=0.004468684550374746, zero_point=151)
  (quant): Quantize(scale=tensor([0.0039]), zero_point=tensor([0]), dtype=torch.quint8)
  (dequant): DeQuantize()
)

MUD graph graph(%self : __torch__.torch.testing._internal.common_quantization.___torch_mangle_3.ConvModel,
      %x.1 : Tensor):
  %4 : None = prim::Constant()
  %2 : __torch__.torch.nn.modules.conv.___torch_mangle_2.ConvTranspose2d = prim::GetAttr[name="conv"](%self)
  %x.3 : Tensor = prim::CallMethod[name="forward"](%2, %x.1, %4) # /home/zafar/Git/pytorch-dev/pytorch-convtranspose-new/torch/testing/_internal/common_quantization.py:594:12
  return (%x.3)

quantized MUD graph graph(%self : __torch__.torch.testing._internal.common_quantization.___torch_mangle_18.ConvModel,
      %x.1 : Tensor):
  %5 : int = prim::Constant[value=1]() # /home/zafar/Git/pytorch-dev/pytorch-convtranspose-new/torch/nn/modules/conv.py:909:66
  %4 : int[] = prim::Constant[value=[0, 0]]()
  %3 : int[] = prim::Constant[value=[1, 1]]()
  %self.conv.bias : None = prim::Constant()
  %self.conv.weight : Float(3:45, 5:9, 3:3, 3:1, requires_grad=0, device=cpu) = prim::Constant[value=<Tensor>]()
  %x.3 : Tensor = aten::conv_transpose2d(%x.1, %self.conv.weight, %self.conv.bias, %3, %4, %4, %5, %3) # /home/zafar/Git/pytorch-dev/pytorch-convtranspose-new/torch/nn/modules/conv.py:911:15
  return (%x.3)
```

z-a-f replied:
#resolved Thanks Jerry!

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Looks good

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

vkuzo commented:
would be great to include some context and a test plan in the PR summary

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]

Commit:
Update on "[quant] conv_transpose graph patterns"


Differential Revision: [D23228954](https://our.internmc.facebook.com/intern/diff/D23228954)

[ghstack-poisoned]
